File: app/src/main/java/com/george/android/tasker/data/notes/main_notes/NoteDao.java
package com.george.android.tasker.data.notes.main_notes;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface NoteDao {

    @Insert
    void insert(Note note);

    @Update
    void update(Note note);

    @Delete
    void delete(Note note);

    @Query("DELETE FROM note_table")
    void deleteAllNotes();

    @Query("SELECT * FROM note_table")
    LiveData<List<Note>> getAllNotes();

    @Query("SELECT * FROM note_table WHERE title LIKE :search OR description LIKE :search")
    LiveData<List<Note>> findNote(String search);

}

File: kotlin_voltage_offline/src/main/java/com/george/android/kotlin_voltage_offline/data/database/dao/NoteDao.kt
package com.george.android.kotlin_voltage_offline.data.database.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import com.george.android.kotlin_voltage_offline.data.model.Note

@Dao
interface NoteDao {

    @Insert
    fun insert(note: Note)

    @Update
    fun update(note: Note)

    @Query("DELETE FROM note_table WHERE id LIKE :noteId")
    fun delete(noteId: Int)

    @Query("SELECT * FROM note_table")
    fun getAllNotes(): LiveData<List<Note>>

}